.github/workflows/ci_action.yml

```diff
@@ -90,6 +90,9 @@ jobs:
           pip install -e .[DEV,ML]
           pip install gdal==$(gdal-config --version)
 
+      - name: Set up local cluster
+        run: ray start --head
```

Comment:
You mentioned issues with initializing ray programmatically from python with `ray.init()`. Is that still relevant and would that not be the preferred way of running things on a local ray cluster by default?

Or is this exact logic implemented, and you're just making sure the cluster is set up separately for github actions?

Reply:
> IMO the cluster should be set up in whatever way the user wants,

The user wants it to be a black box 😈 

In all seriousness, it just feels that doing `ray start --head` every time you want to run something is a bit cumbersome for the users. Doing it automatically under the hood is more in line with the current user experience, but I agree that having the option for the user to start the cluster manually should also be supported.

[According to ray](https://docs.ray.io/en/latest/ray-core/starting-ray.html#starting-ray-on-a-single-machine) it seems a completely legit way of initializing the cluster.

Perhaps what I'm suggesting is not the best solution, since we have to try first how this change will behave anyway, but in that case maybe we're missing some instructions what the user should do in order to start the cluster, if the cluster can't be found

Reply:
from what we experimented with back in the day using `ray.init()` did not connect you to the remote cluster, it just created a local one. We had to add the `address="auto"` which then blocks if it cannot find a running cluster.

And judging by the documentation that you linked, nothing has changed.

Since eo-grow is for grown-ups (and very talented younglings) i think they can manage a `ray start --head` once a day when running things locally :P

Reply:
i re-read the whole docs, and it might be that it works now. I would still prefer for the user to spin up the cluster themselves, but you can create a new MR and test out the possible scenarios and if all work, i am willing to merge it.